src/lib/import-export/import/import-manager.ts

```diff
@@ -0,0 +1,51 @@
+import fsPromises from 'fs/promises';
+import os from 'os';
+import path from 'path';
+import { BackupHandlerFactory } from './handlers/backup-handler-factory';
+import { Importer, ImporterResult } from './importers/Importer';
+import { BackupArchiveInfo, BackupContents } from './types';
+import { Validator } from './validators/Validator';
+
+export function selectImporter(
+	allFiles: string[],
+	extractionDirectory: string,
+	validators: Validator[],
+	importers: { [ key: string ]: new ( backup: BackupContents ) => Importer }
+): Importer | null {
+	for ( const validator of validators ) {
+		if ( validator.canHandle( allFiles ) ) {
+			const importerClass = importers[ validator.constructor.name ];
+			if ( importerClass ) {
+				const files = validator.parseBackupContents( allFiles, extractionDirectory );
+				return new importerClass( files );
+			}
+		}
+	}
+	return null;
+}
+
+export async function importBackup(
+	backupFile: BackupArchiveInfo,
+	sitePath: string,
+	validators: Validator[],
+	importers: { [ key: string ]: new ( backup: BackupContents ) => Importer }
+): Promise< ImporterResult > {
+	const extractionDirectory = await fsPromises.mkdtemp( path.join( os.tmpdir(), 'studio_backup' ) );
```

Comment:
I'd suggest including a UUID in the temporary path to allow multiple import processes simultaneously.

Reply:
You mean an extra id?
Because from the docs of `mkdtemp`:
```
The fs.mkdtemp() method is used to create a unique temporary directory. The folder name is created by appending 6 randomly generated characters behind a prefix string. It is also possible to create the temporary directory inside a folder by using a separator after the folder path. ```

Reply:
Ah, true. I overlooked that function `mkdtemp` already appends a random id. I confused this with another implementation I did in E2E tests where I appended a UUID for a temp folder ([reference](https://github.com/Automattic/studio/blob/ca29e98b1aa7f28210b93cc2b5c9c6c11e96bc2e/e2e/e2e-helpers.ts#L18)). Forget my comment 😅 .